playground/backend-integration/__tests__/backend-integration.spec.ts

```diff
@@ -26,9 +23,26 @@ test('should have no 404s', () => {
 
 describe('asset imports from js', () => {
   test('file outside root', async () => {
-    expect(
-      await page.textContent('.asset-reference.outside-root .asset-url'),
-    ).toMatch(outerAssetMatch)
+    await vi.waitFor(async () => {
```

Comment:
Question: Do we need to `waitFor` all the expects inside here? Or could we wait for something specific instead?

Reply:
Since it depends on client script execution, we probably should wait for something (thought it wasn't so previously). I think I can move up the image assertion first to avoid wrapping entire things.
Updated in https://github.com/vitejs/vite/pull/17886/commits/8d1715beea14cd95c1dcbfc4465c706f654780e4